### DISCUSSION TITLE
I didn't find how to get font height.

### DISCUSSION THREAD

**Discussion by totake0224** [harfbuzz/harfbuzz#4468] [Q&A] [answered]
I want to get font height that set fontforge's font information window.
I tried hb_ot_metrics_get_position and hb_face_get_upem function but I didn't get the value.
please tell me  is there another function that I got font informations.

**Comment by behdad** [comment]
Try `hb_font_get_extents`.

**Comment by totake0224** [comment]
Thank you.
I couldn't find hb_font_get_extents so I tried hb_font_get_extents_for_direction. but I got same value of 
	hb_ot_metrics_get_position(hb_font , HB_OT_METRICS_TAG_HORIZONTAL_ASCENDER,&ascender);
	hb_ot_metrics_get_position(hb_font , HB_OT_METRICS_TAG_HORIZONTAL_DESCENDER,&descender);
I think I have to check fontforge source to find which parameter it shows. but it seems difficult.

**Reply by behdad** [reply]
Did you check the extra `line_gap`?

**Reply by totake0224** [reply]
Hello.
Yes , line_gap is 0.
I tried HB_DIRECTION_LTR,HB_DIRECTION_RTL, HB_DIRECTION_TTB for hb_ot_metrics_get_position
and all of hb_ot_metrics_tag_t (28 enums) for hb_ot_metrics_get_position.
I attached font file that I tried to get height parameter.
fontforge shows it's height 880. but I could not get that value from harfbuz function.
[_MC-Test.zip](https://github.com/harfbuzz/harfbuzz/files/13225314/_MC-Test.zip)

**Reply by behdad** [reply, marked as answer]
The 880 is the `sTypoAscender` in the `OS/2` table. However the `OS/2.fsSelection` doesn't have the bit for `USE_TYPO_METRICS` set. So we get the ascender from the `hhea` table, that has 1024.

You can use some private tags with `hb-ot-metrics` API to get the value you want:
```
#define _HB_OT_METRICS_TAG_HORIZONTAL_ASCENDER_OS2   HB_TAG ('O','a','s','c') 
#define _HB_OT_METRICS_TAG_HORIZONTAL_ASCENDER_HHEA  HB_TAG ('H','a','s','c') 
#define _HB_OT_METRICS_TAG_HORIZONTAL_DESCENDER_OS2  HB_TAG ('O','d','s','c') 
#define _HB_OT_METRICS_TAG_HORIZONTAL_DESCENDER_HHEA HB_TAG ('H','d','s','c') 
#define _HB_OT_METRICS_TAG_HORIZONTAL_LINE_GAP_OS2   HB_TAG ('O','l','g','p') 
#define _HB_OT_METRICS_TAG_HORIZONTAL_LINE_GAP_HHEA  HB_TAG ('H','l','g','p') 
```

**Reply by totake0224** [reply]
Thank you.
I got fontforge height value from _HB_OT_METRICS_TAG_HORIZONTAL_ASCENDER_OS2.
You are great !!
I will be very grad if you tell me how to identify these information.

**Reply by totake0224** [reply]
I'm sorry it's my mistake.  _HB_OT_METRICS_TAG_HORIZONTAL_ASCENDER_HHEA is a right parameter for the height.
